src/array/cuda/spmat_op_impl_csr.cu

```diff
@@ -415,25 +405,33 @@ template std::vector<NDArray> CSRGetDataAndIndices<kDLGPU, int64_t>(
  */
 template <typename IdType>
 __global__ void _SegmentMaskColKernel(
-    const IdType* indptr, const IdType* indices, int64_t num_rows,
+    const IdType* indptr, const IdType* indices, int64_t num_rows, int64_t num_nnz,
     const IdType* col, int64_t col_len,
     IdType* mask, IdType* count) {
+  IdType tx = static_cast<IdType>(blockIdx.x) * blockDim.x + threadIdx.x;
+  const int stride_x = gridDim.x * blockDim.x;
+  while (tx < num_nnz) {
+    IdType rpos = dgl::cuda::_UpperBound(indptr, num_rows, tx) - 1;
+    IdType cur_c = indices[tx];
+    IdType i = dgl::cuda::_BinarySearch(col, col_len-1, cur_c);
+    if (i < col_len) {
+      mask[tx] = 1;
+      cuda::AtomicAdd(count+rpos, IdType(1));
+    }
+    tx += stride_x;
+  }
+}
+
+template <typename IdType>
+__global__ void _CheckSorted(const IdType* A, int64_t n, int8_t *flags) {
   int tx = blockIdx.x * blockDim.x + threadIdx.x;
   const int stride_x = gridDim.x * blockDim.x;
-  // TODO(minjie): consider putting the col array in shared memory.
-  while (tx < num_rows) {
-    IdType cnt = 0;
-    for (IdType i = indptr[tx]; i < indptr[tx + 1]; ++i) {
-      const IdType cur_c = indices[i];
-      for (int64_t j = 0; j < col_len; ++j) {
-        if (cur_c == col[j]) {
-          mask[i] = 1;
-          ++cnt;
-          break;
-        }
-      }
+  while (tx < n-1) {
+    bool f = true;
+    if (A[tx] > A[tx+1]) {
+      f = false;
     }
-    count[tx] = cnt;
+    flags[tx] = f;
```

Comment:
It should compile to the same, but can we simplify this to `flags[tx] = A[tx] <= A[tx+1]`?